clean_air/util/dataframe_converter.py

```diff
@@ -0,0 +1,148 @@
+"""
+This module contains functions to convert between dataframe types.
+"""
+
+from clean_air.util.cubes import get_xy_coords
+from iris.pandas import _assert_shared, _as_pandas_coord, as_data_frame
+import numpy as np
+import pandas as pd
+from geopandas import GeoDataFrame
+from shapely.geometry import Point
+import warnings
+
+
+def as_pandas_data(data, copy=True):
+    """
+    Sort iris cube style data into pandas style data.
+    :param data: Data array to be converted.
+    :param copy: Boolean value representing whether the data should be copied.
+    or not (this will return an error if the array is masked).
+    :return: Data array in pandas format.
+    """
+    if isinstance(data, np.ma.MaskedArray):
+        if not copy:
+            raise ValueError("Masked arrays must always be copied.")
+        data = data.astype('f').filled(np.nan)
+    elif copy:
+        data = data.copy()
+
+    return data
+
+
+def as_series(cube, index=None, copy=True):
+    """
+    Static function to convert an iris cube into a pandas series.
+
+    :param cube: iris cube (1-dimensional) for conversion.
+    :param index: 1-dimensional array representing the order of data for input
+    (if not supplied, the input order will be the same as the cube data).
+    :param copy: boolean value representing whether to keep a copy of the
+    data upon conversion.
+    :return: pandas Series containing information converted from iris cube.
+    """
+    # NOTE: This is not yet used in this converter, but it will be needed soon.
+    # It is as yet untested and unfinished; pulled from iris but in need of
+    # simplification and modification.
+    data = as_pandas_data(cube.data)
+    if not index:
```

Comment:
Rightio, I will change this too.  Had spotted some indexy things that I was uncertain about, but somehow thought that iris would have got it right.  Turns out they've done a pretty bad job of all the pandas stuff though.

Reply:
Still feel that I should be changing more of this function, especially since I have now changed the comment to reflect the fact that this IS now used and tested, but it's not very tested and I still don't feel comfortable with it.

For example, there's the `_assert_shared` bit which I don't like; I know that it's something to do with how pandas deals with masked data, but I don't really know what it does and that makes me very uncomfortable.